### PR TITLE
Check and fix non-inclusive laguange

### PR DESCRIPTION
**Pull request by daniloegea**
## Description
This PR introduces a new github action used to check for non-inclusive wording in our code base and documentation using Woke.

While not all the occurrences of non-inclusive words were fixed, it also replaces many occurrences with proper language. Most of the remaining occurrences are due to external APIs and tools we rely on.

It provides alternative names for the options below:
    all-slaves-active -> all-members-active
    packets-per-slave -> packets-per-member

The yaml emitter will produce yaml documents using the new words.

Ideally, every single unit test containing the old words should be duplicated to also handle the new one, but at the moment only two tests were adapted.

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [x] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

### CODE REVIEW

**Comment by sil2100** [comment]
An initial thought when looking at the non-inclusive language check: the output from the test run isn't very 'useful'. Is that how the standard woke command reports warnings and errors? Since when looking at the CI run, I only see mentions insensitive words and its alternatives, but doesn't quite point to where the string appears in code. Would it be possible to make it pinpoint the location? I must say I never used `woke` myself ever.

Also, I think we'll have to wait with a final review of this card until Lukas and Steve are back as this basically requires changing the netplan schema, so we need to have an overall agreement on what we want it to be.

File: tests/integration/wifi.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ class TestNetworkManager(IntegrationTestsWifi, _CommonTests):
 
     def test_wifi_ap_open(self):
         # we use dev_w_client and dev_w_ap in switched roles here, to keep the
-        # existing device blacklisting in NM; i. e. dev_w_client is the
+        # existing device blocklisting in NM; i. e. dev_w_client is the
```
Comment (slyon) [inline tests/integration/wifi.py:155]: nitpick: I'd prefer `denylisting` here, "blocklisting" makes me think about a block of devices.

File: tests/generator/test_passthrough.py [hunk 2]
```diff
@@ -122,22 +122,22 @@ def test_passthrough_type_nm_devices(self):
       networkmanager:
         passthrough:
           connection.uuid: 87749f1d-334f-40b2-98d4-55db58965f5f
-          connection.type: dummy''')
```
Comment (slyon) [inline tests/generator/test_passthrough.py:125]: We should keep upstream's `dummy` type here: https://developer-old.gnome.org/NetworkManager/stable/settings-dummy.html

File: src/abi.h [hunk 3]
```diff
@@ -269,6 +269,7 @@ struct netplan_net_definition {
         char* transmit_hash_policy;
         char* selection_logic;
         gboolean all_slaves_active;
+#define all_members_active all_slaves_active
```
Comment (slyon) [inline src/abi.h:272]: I feel like we should rather rename the struct field and make use of the new `all_members_active` variable name in our internal code. Same goes for `packets_per_member` below.

In any case, I'd make the new, inclusive name the default variable name and have the old, non-inclusive name be the `#define`/alias.

File: src/parse.c [hunk 4]
```diff
@@ -2112,6 +2112,7 @@ static const mapping_entry_handler bond_params_handlers[] = {
     {"transmit-hash-policy", YAML_SCALAR_NODE, {.generic=handle_netdef_str}, netdef_offset(bond_params.transmit_hash_policy)},
     {"ad-select", YAML_SCALAR_NODE, {.generic=handle_netdef_str}, netdef_offset(bond_params.selection_logic)},
     {"all-slaves-active", YAML_SCALAR_NODE, {.generic=handle_netdef_bool}, netdef_offset(bond_params.all_slaves_active)},
+    {"all-members-active", YAML_SCALAR_NODE, {.generic=handle_netdef_bool}, netdef_offset(bond_params.all_members_active)},
```
Comment (slyon) [inline src/parse.c:2115]: The new/alternative "all-members-active" YAML stanza works for me, but let's also try to get Steve's opinion.

But I think I'd stick to the new, inclusive name for our internal variable naming.
```suggestion
    {"all-slaves-active", YAML_SCALAR_NODE, {.generic=handle_netdef_bool}, netdef_offset(bond_params.all_members_active)},
    {"all-members-active", YAML_SCALAR_NODE, {.generic=handle_netdef_bool}, netdef_offset(bond_params.all_members_active)},
```

File: src/parse.c [hunk 5]
```diff
@@ -2125,6 +2126,7 @@ static const mapping_entry_handler bond_params_handlers[] = {
     {"gratuitious-arp", YAML_SCALAR_NODE, {.generic=handle_netdef_guint}, netdef_offset(bond_params.gratuitous_arp)},
     /* TODO: unsolicited_na */
     {"packets-per-slave", YAML_SCALAR_NODE, {.generic=handle_netdef_guint}, netdef_offset(bond_params.packets_per_slave)},
+    {"packets-per-member", YAML_SCALAR_NODE, {.generic=handle_netdef_guint}, netdef_offset(bond_params.packets_per_member)},
```
Comment (slyon) [inline src/parse.c:2129]: The new/alternative "packets-per-member" YAML stanza works for me, but let's also try to get Steve's opinion.

But I think I'd stick to the new, inclusive name for our internal variable naming.
```suggestion
    {"packets-per-slave", YAML_SCALAR_NODE, {.generic=handle_netdef_guint}, netdef_offset(bond_params.packets_per_member)},
    {"packets-per-member", YAML_SCALAR_NODE, {.generic=handle_netdef_guint}, netdef_offset(bond_params.packets_per_member)},
```

File: doc/netplan-yaml.md [hunk 6]
```diff
@@ -1144,7 +1144,7 @@ wpasupplicant installed if you let the `networkd` renderer handle wifi.
     > `bandwidth`, and `count`. This option is only used in 802.3ad
     > mode.
 
-  - **all-slaves-active** (bool)
+  - **all-members-active**, **all-slaves-active** (bool)
```
Comment (slyon) [inline doc/netplan-yaml.md:1147]: I'd rather put the "all-slaves-active" title down into the description, e.g. as `Alias: **all-slaves-active**`

File: doc/netplan-yaml.md [hunk 7]
```diff
@@ -1199,41 +1199,41 @@ wpasupplicant installed if you let the `networkd` renderer handle wifi.
...
+    > up on a new port, a notification is sent and possibly repeated if
     > this value is set to a number greater than `1`. The default value
     > is `1` and valid values are between `1` and `255`. This only
     > affects `active-backup` mode.
     >
     > For historical reasons, the misspelling `gratuitious-arp` is also
     > accepted and has the same function.
 
-  - **packets-per-slave** (scalar)
+  - **packets-per-member**, **packets-per-slave** (scalar)
```
Comment (slyon) [inline doc/netplan-yaml.md:1217]: I'd rather put the "packets-per-slave" title down into the description, e.g. as Alias: **packets-per-slave**

File: tests/integration/base.py [hunk 8]
```diff
@@ -443,7 +443,7 @@ def create_devices(klass):
         klass.dev_w_client = devs[1]
 
         # don't let NM trample over our fake AP
-        with open('/run/NetworkManager/conf.d/test-blacklist.conf', 'w') as f:
+        with open('/run/NetworkManager/conf.d/test-blocklist.conf', 'w') as f:
```
Comment (slyon) [inline tests/integration/base.py:446]: nitpick: I'd prefer denylist instead of blocklist (also above).

**Review by slyon — CHANGES_REQUESTED** [review]
Thanks for setting up the new CI hook!

I think we should make use of `# wokelint:rule=...` comments throughout the code, to ignore exception (i.e. the things we cannot change).

Most cases are inside the tests, where we should be able to apply some search & replace logic to catch many cases at once. Maybe we need to adopt the `assert_nm` method in `tests/generator/base.py` to ignore `# wakeignore: ...` comments inside the provided keyfile.

For (internal) vairable naming, I'd not keep the old name around, but just rename the variable or struct fields accordingly, this should solve another class of warnings.

File: src/abi.h (hunk 3 above)
Comment (daniloegea) [reply src/abi.h:272]: I'll go with renaming them

File: .woke.yaml [hunk 9]
```diff
@@ -0,0 +1,187 @@
+ignore_files:
+  - abi-compat/*.xml
+  - tests/generator/test_vrfs.py
+  - tests/generator/test_passthrough.py
+  - tests/parser/test_keyfile.py
+  - tests/generator/test_bridges.py
+  - tests/generator/test_bonds.py
+  - tests/generator/test_tunnels.py
+  - tests/generator/test_common.py
```
Comment (slyon) [inline .woke.yaml:9]: I thought about ignoring some of the tests as well (for now). But OTOH, now is the time to fix everything `woke` finds... It's only ~60 lines which need fixing left, so I think we should put in the extra effort to fix those instances as well.

We might need to adopt the `assert_nm()`, `assert_networkd()` and `generate_keyfile()` methods in their corresponding `base.py` class, but I think it's worth the effort. Like having a simple RegEx string-replace to search for something like `  # wokeignore:rule...` and replace it with an empty string in the corresponding keyfile/networkd assertion function, before comparing it to the actually generated NM/networkd config. This way we should be able to put our "wokeignore" comments into the test's code, while staying compatible with the generated output.

**Review by slyon — CHANGES_REQUESTED** [review]
Thank you very much for addressing the previous comments and adding a bunch of ignore-rules at the relevant places.

I left some more inline comments, which we could improve upon, while also waiting for @vorlonofportland 's opinion on the new **all-members-active** and **packets-per-member** stanzas (alias).

Generally, I wonder where we are supposed to get our woke ruleset from (and how to keep it updated), besides woke's default ruleset. I see you picked some additional rules from https://inclusivenaming.org and from https://github.com/canonical/Inclusive-naming How did you get to pick those? Is there any policy around this?

Another topic is about the `blocklist` naming, which I'm personally not a big fan of, as I tend to read "block" as a chunk of something (i.e. a "block of code"). I'd prefer the allow/deny terminology. But I see there's ongoing discussions on this topic also in https://github.com/canonical/apport/pull/58

**Comment by vorlonofportland** [comment]
These schema changes look appropriate to me.  The one thing I note is that the underlying backends (NM, networkd, and the kernel itself) still use the "slave" language.  Will there be follow-up to try to promulgate this name change down the stack?

**Comment by slyon** [comment]
> The one thing I note is that the underlying backends (NM, networkd, and the kernel itself) still use the "slave" language. Will there be follow-up to try to promulgate this name change down the stack?

ACK. I think it would make sense to cross-reference existing "inclusive-naming" bug reports in the corresponding backends, if already existing, or create such bug reports upstream.

File: doc/netplan-yaml.md [hunk 10]
```diff
@@ -1144,12 +1144,12 @@ wpasupplicant installed if you let the `networkd` renderer handle wifi.
...
     > mode.
 
-  - **all-slaves-active** (bool)
+  - **all-members-active** (bool)
 
     > If the bond should drop duplicate frames received on inactive ports,
     > set this option to `false`. If they should be delivered, set this
     > option to `true`. The default value is false, and is the desirable
-    > behavior in most situations.
+    > behavior in most situations. Alias: **all-slaves-active**  <!--- wokeignore:rule=slave -->
```
[the remark this answers is not in the document]
Comment (slyon) [reply doc/netplan-yaml.md]: We should keep the "Alias" description as part of the same "quotation" block, IMO (e.g. using the `    > ` prefix)

File: tests/integration/base.py (hunk 8 above)
Comment (slyon) [reply tests/integration/base.py]: There are more instances of the `test-blocklist.conf` below in `tearDownClass()`.

File: tests/generator/test_bonds.py [hunk 11]
```diff
@@ -197,7 +197,82 @@ def test_bond_with_parameters(self):
...
+        down-delay: 30
+        all-slaves-active: true  # wokeignore:rule=slave
+        transmit-hash-policy: none
+        ad-select: none
+        arp-interval: 15
+        arp-validate: all
+        arp-all-targets: all
+        fail-over-mac-policy: none
+        gratuitious-arp: 10
+        packets-per-slave: 10  # wokeignore:rule=slave
```
Comment (sil2100) [inline tests/generator/test_bonds.py:246]: Should we also simply use `packets-per-member` here as well?

File: tests/generator/test_bonds.py [hunk 12]
```diff
@@ -197,7 +197,82 @@ def test_bond_with_parameters(self):
...
+  bonds:
+    bn0:
+      parameters:
+        mode: 802.3ad
+        lacp-rate: 10
+        mii-monitor-interval: 10
+        min-links: 10
+        up-delay: 20
+        down-delay: 30
+        all-slaves-active: true  # wokeignore:rule=slave
```
Comment (sil2100) [inline tests/generator/test_bonds.py:238]: Same here, maybe `all-members-active`?

**Review by sil2100 — APPROVED** [review]
This looks good now, I see all comments from Lukas seem to be resolved. There's still 2 lines I'd like to change, but those are nitpicks.

File: tests/generator/test_bonds.py (hunk 11 above)
Comment (daniloegea) [reply tests/generator/test_bonds.py:246]: These instances were left there on purpose. As these are valid configuration we can't remove just yet, we should have tests making use of them.

**Comment by slyon** [comment]
I created some upstream RFC's about using inclusive language in our backends networkd and NetworkManager:

* https://github.com/systemd/systemd/issues/26257
* https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/1205

The Open vSwitch did the change already in v2.15:
* https://github.com/openvswitch/ovs/commit/91fc374a9c5a2a4d9520aaa3588a7a18338a476e

**Pull request merged**